src/metabase/native_query_analyzer/replacement.clj

```diff
@@ -44,21 +51,33 @@
       (string? token)
       (recur rest (str query-so-far token) substitutions)
 
-      (params/Param? token)
-      (let [sub (sentinel-variable raw-query)]
-        (recur rest
-               (str query-so-far sub)
-               (add-tag substitutions sub token)))
+      :else
+      (let [v         (params->values (:k token))
```

Comment:
It seems like this map goes from a single param to a single value? The name suggests that it takes and returns collections, or at best is ambiguous. I haven't down into the actual shape of this map, so excuse any faulty speculation.

```suggestion
      (let [v         (param->value (:k token))
```

Reply:
:heavy_check_mark: